File: Lab-07-solution/src/main/java/com/petclinic/core/Visit.java
package com.petclinic.core;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.CascadeType;

import java.time.LocalDate;

@Entity
public class Visit {
    @Id @GeneratedValue
    private Long id;

    @OneToOne @Cascade(CascadeType.ALL)
    private Pet pet;

    @OneToOne @Cascade(CascadeType.ALL)
    private Owner owner;

    private String referenceNumber;
    private LocalDate date;
    private String purpose;



    public Visit() {
    }

    public Visit(Long id, String referenceNumber, LocalDate date, String purpose) {
        this.id = id;
        this.referenceNumber = referenceNumber;
        this.date = date;
        this.purpose = purpose;
    }

    public Long getId() {
        return id;
    }

    public String getReferenceNumber() {
        return referenceNumber;
    }

    public LocalDate getDate() {
        return date;
    }

    public String getPurpose() {
        return purpose;
    }

    public void setPet(Pet pet) {
        this.pet = pet;
    }
    public void setOwner(Owner owner) {
        this.owner = owner;
    }

    public Pet getPet() {
        return pet;
    }

    public Owner getOwner() {
        return owner;
    }
}

File: Lab-kotlin/src/main/kotlin/com/petclinic/core/Visit.kt
package com.petclinic.core

import jakarta.persistence.*
import java.time.LocalDate

@Entity
class Visit(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    var date: LocalDate,
    var purpose: String,

    @OneToOne(cascade = [CascadeType.PERSIST])
    @JoinColumn(name="owner_id")
    var owner: Owner?,
    @OneToOne(cascade = [CascadeType.PERSIST])
    @JoinColumn(name="pet_id")
    var pet: Pet?
)
